src/cmd/manage_nsfs.js

```diff
@@ -748,6 +758,42 @@ async function validate_account_args(data, action) {
     }
 }
 
+/** validate_options checks if there are options that are not valid.
+ * valid options are: required arguments, optional options and global configurations.
+ * @param {string} type
+ * @param {string} action
+ * @param {object} argv
+ */
+function validate_options(type, action, argv) {
+    if (!Object.values(TYPES).includes(type)) throw_cli_error(ManageCLIError.InvalidType);
+    if (type === TYPES.ACCOUNT || type === TYPES.BUCKET) {
+        if (!Object.values(ACTIONS).includes(action)) throw_cli_error(ManageCLIError.InvalidAction);
+    } else if (type === TYPES.IP_WHITELIST) {
+        if (action !== '') throw_cli_error(ManageCLIError.InvalidAction);
+    }
+
+    const input_options = Object.keys(argv); // we don't care about the value, only the flags
+    input_options.shift(); // the first element is _ with the type and action, so we remove it
+
+    let valid_options;
+    if (type === TYPES.BUCKET) {
+        valid_options = VALID_OPTIONS.bucket_options[action];
+    } else if (type === TYPES.ACCOUNT) {
+        valid_options = VALID_OPTIONS.account_options[action];
+    } else {
+        valid_options = VALID_OPTIONS.wihtelist_options;
+    }
+
+    valid_options = new Set(valid_options); // for performance, we use Set as data structure
```

Comment:
you can use Set() at the definition of the valid options in manafe_nsfs_constants
the additional iteration on the array for creating the Set() decreases the performance unwillingly

Reply:
Thank you, this what I wished, but missed it.